csrc/executor_utils.cpp

```diff
@@ -1151,6 +1151,9 @@ CompiledKernel::~CompiledKernel() {
   if (module != nullptr) {
     NVFUSER_CUDA_SAFE_CALL(cuModuleUnload(module));
   }
+#ifndef NDEBUG
```

Comment:
if (module != nullptr) {
  unload
  module = nullptr;
}

Also, I'd remove NDEBUG unless you find it indeed expensive to set null.